airflow/providers/jenkins/operators/jenkins_job_trigger.py

```diff
@@ -153,21 +153,30 @@ def poll_job_in_queue(self, location: str, jenkins_server: Jenkins) -> int:
         # once it will be available in python-jenkins (v > 0.4.15)
         self.log.info('Polling jenkins queue at the url %s', location)
         while try_count < self.max_try_before_job_appears:
-            location_answer = jenkins_request_with_headers(
-                jenkins_server, Request(method='POST', url=location)
-            )
-            if location_answer is not None:
-                json_response = json.loads(location_answer['body'])
-                if (
-                    'executable' in json_response
-                    and json_response['executable'] is not None
-                    and 'number' in json_response['executable']
-                ):
-                    build_number = json_response['executable']['number']
-                    self.log.info('Job executed on Jenkins side with the build number %s', build_number)
-                    return build_number
-            try_count += 1
-            time.sleep(self.sleep_time)
+            try:
+                location_answer = jenkins_request_with_headers(
+                    jenkins_server, Request(method='POST', url=location)
+                )
+                if location_answer is not None:
+                    json_response = json.loads(location_answer['body'])
+                    if (
+                        'executable' in json_response
+                        and json_response['executable'] is not None
+                        and 'number' in json_response['executable']
+                    ):
+                        build_number = json_response['executable']['number']
+                        self.log.info('Job executed on Jenkins side with the build number %s', build_number)
+                        return build_number
+                try_count += 1
+                time.sleep(self.sleep_time)
+
+            # we don't want to fail the operator, this will continue to poll
+            # until max_try_before_job_appears reached
+            except (HTTPError, JenkinsException) as ex:
```

Comment:
When are these exceptions raised, respectively? This try-catch block seems too big, a lot of the wrapped code is response processing that should be outside of this block, from what I can tell.

Reply:
you are right it was a little careless to include everything in try except, 
the only part raising exceptions is the call to `jenkins_request_with_headers` method, 
I changed the code